api/envoy/extensions/filters/common/set_filter_state/v3/value.proto

```diff
@@ -35,6 +35,9 @@ message FilterStateValue {
     // Filter state object key. The key is expected to be registered via an object factory, see
     // :ref:`the well-known filter state keys <well_known_filter_state>`.
     string object_key = 1 [(validate.rules).string = {min_len: 1}];
+
+    // Use a string value object to store the value under the given key.
+    string string_key = 6 [(validate.rules).string = {min_len: 1}];
```

Comment:
Compare to providing a new `string_key`, I think it would be better to provide a new `factory_key` which will be used to find the object factory. And if the `factory_key` is empty, we will try the `object_key` to find a factory.

Then we need to create a unique factory for the `Router::StringAccessorImpl`.

If an object which will store raw string value is necessary, we just need to set the `factory_key` to the factory name of `Router::StringAccessorImpl`.

And this way could split the object name and factory name.

Reply:
cc @lizan

Reply:
I'm not sure it's a good idea to separate the key name from the factory name. The filter state objects are typically distinct for different names, it's not a good idea to allow putting an object from key A into key B since filters A and B do not share their objects. 

The string accessor is used widely and it useful to avoid compiling in a definition of an object factory. What I mean to say is that if we add `factory_key` then the only recommended value would be this string accessor factory. Anything else there would likely create issues.

Reply:
> I'm not sure it's a good idea to separate the key name from the factory name. The filter state objects are typically distinct for different names, it's not a good idea to allow putting an object from key A into key B since filters A and B do not share their objects.

Hmmm, basically, I think type and the key shouldn't be bound strictly. And in our actual practices, we use a common struct/type to store value from different filter with different key. So, I think this should be a reasonable requirement.

> The string accessor is used widely and it useful to avoid compiling in a definition of an object factory. What I mean to say is that if we add factory_key then the only recommended value would be this string accessor factory. Anything else there would likely create issues.

I think there also are uint32/uint64 accessor that may be necessary for users. But I think we will don't expect to add new `integer_key` in the future.

Reply:
OK, I think I like your idea because we won't have to change the xDS API in the future if we use `factory_key` to designate a different constructor.